x-pack/test/functional/services/ml/anomaly_explorer.ts

```diff
@@ -66,5 +67,20 @@ export function MachineLearningAnomalyExplorerProvider({ getService }: FtrProvid
     async assertSwimlaneViewByExists() {
       await testSubjects.existOrFail('mlAnomalyExplorerSwimlaneViewBy');
     },
+
+    async openAddToDashboardControl() {
+      await testSubjects.click('mlAnomalyTimelinePanelMenu');
+      await testSubjects.click('mlAnomalyTimelinePanelAddToDashboardButton');
+      await testSubjects.existOrFail('mlAddToDashboardModal');
+    },
+
+    async addAndEditSwimlaneInDashboard(dashboardId: string) {
+      await testSubjects.isDisplayed('mlDashboardSelection');
+      await testSubjects.isDisplayed(`checkboxSelectRow-${dashboardId}`);
+      await testSubjects.click(`checkboxSelectRow-${dashboardId}`);
+      await testSubjects.clickWhenNotDisabled('mlAddAndEditDashboardButton');
+      const swimlane = await find.byClassName('ml-swimlanes');
```

Comment:
This check doesn't make sure that we're actually viewing the dashboard. In case the button would just close the modal, we would still get a success here since the anomaly explorer itself displays the same swimlanes.
I'd suggest to add a ```data-test-subj={`ml-swimlane-${swimlaneType}`}``` to the [swimlane div](https://github.com/elastic/kibana/blob/master/x-pack/plugins/ml/public/application/explorer/explorer_swimlane.tsx#L638) that already has the class you're using. Then we could go with e.g. `testSubjects.isDisplayed('dashboardPanel > ml-swimlane-oveall')`, which should probably be wrapped in a retry to allow some dashboard loading time (might be even worth a separate service method).

Reply:
Fixed in https://github.com/elastic/kibana/pull/68784/commits/c509d2f647dde130baf298d64fc3d3cb2eb61726